examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,274 @@
+# dimensions. We explore only a few aspects of the algorithm but our general
+# evaluation function can be used to dig deeper in the algo.
+#
+# Notice that we propose to vary several aspects:
+# * `n_samples` is self explanatory
+# * `n_repeat` is the number of repetition of the experiment (to see the
+#    result variance).
+# * `n_features` is the number of dimensions
+# * `n_blobs` is the number of clusters in the ground truth
+# * `n_clusters` is the number of clusters KMeans is looking for
+# * `n_fixed_clusters` is the number of clusters we keep fixed
+# * `reassignement_ratio` is a parameter of the MiniBatchKMeans. It
+#   indicates the ratio of centers of high inertia reassigned randomly
+# * `recenter_every` is a relaxation of the cluster fixation. In our original
+#   version of incremental KMeans, clusters are stuck and never allowed to
+#   move. The recentering parameter allows to let the move, and fix them back
+#   on fixed positions every $k$ iterations. We expect that in some cases, the
+#   fixed cluster could be detrimental to the optimization and thus relaxing
+#   this constraint could lead to a better minimum.
+
+def evaluate(n_samples, n_repeat, n_features, n_blobs, n_clusters,
+             n_fixed_clusters, reassignment_ratio, recenter_every):
+    inertiae = []
+    for i in range(n_repeat):
+        X, y, centers = make_blobs(
+            centers=n_blobs,
+            return_centers=True,
+            n_features=n_features,
+            random_state=i)
+        clus = IncrementalMiniBatchKMeans(
+            n_clusters=n_clusters,
+            reassignment_ratio=reassignment_ratio
+        )
+        kwargs = dict()
+        kwargs['fixed_cluster_centers'] = None
+        if n_fixed_clusters > 0:
+            kwargs['fixed_cluster_centers'] = centers[:n_fixed_clusters]
+        kwargs['recenter_every'] = recenter_every
+        clus.fit(X, **kwargs)
+        inertiae.append(clus.inertia_)
+    return inertiae
+ 
+
+def plot(xlabel, ylabel, values, inertiae):
+    arr = np.asarray(inertiae)
+    for i in range(arr.shape[1]):
+        plt.scatter(values, arr[:, i], alpha=.2, c='gray')
+    plt.plot(values, np.mean(arr, axis=1), c='r')
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.show()
+
+
+##############################################################################
+# Number of fixed cluster centers
+# ===============================
+#
+# In this experiment, we generate data from 10 clusters and see how the number
+# of fixed cluster influence the intertia.
+
+study_value = []
+study_inert = []
+
+for i in range(10):
+    study_value.append(i)
+    inert = evaluate(5000, 100, 20, 10, 10, i, 0.01, None)
+    study_inert.append(inert)
+    
+plot('Number of fixed center clusters /10', 'Mean inertia over 100 runs',
+     study_value, study_inert)
+
+##############################################################################
+# It may seem surprising that the
```

Comment:
Humm I'm not sure what you mean. Isn't it expected as the more you have fixed cluster, the higher the constraints and hence the inertia ?

Reply:
Yes, but I am afraid that some readers may find weird that the true centers do not have best inertia.

Reply:
Fixed in an upcoming commit